### PR TITLE
Add parameters to specify target filename

### PR DESCRIPTION
**Pull request by pothos**
When a match pattern has to be used, the resulting filename in the output directory is unknown.
Add parameters to take only the first match and use a particular name for the extracted target file.

## How to use

Should fix the problem with self-hosted update payloads.

## Testing done

In [scripts PR with Flatcar](https://github.com/flatcar/scripts/pull/1580) and also manual:
```
target/debug/download_sysext -p ~/kinvolk/scripts/sdk_container/src/third_party/coreos-overlay/coreos-base/coreos-au-key/files/official-v2.pub.pem -m "*oem-azure.gz" -o /var/tmp/outdir/ -i /var/tmp/beta-response-local -v -n oem-azure.raw -t
```
where the XML has entries like `<package name="extrafile-amd64-3815.0.0-oem-ami.gz" …`

### CODE REVIEW

File: src/bin/download_sysext.rs [hunk 1]
```diff
@@ -295,6 +295,10 @@ struct Args {
     #[argh(option, short = 'o')]
     output_dir: String,
 
+    /// target filename in directory
```
Comment (dongsupark) [inline src/bin/download_sysext.rs:298]: Ok, the possibility matrix of command-line options is getting bigger and bigger. I am slightly concerned how big it is going to be in the end.

We should add a detailed help message for `--target-filename`, like `this option can be used only together with --take-first-match`.

Comment (pothos) [reply src/bin/download_sysext.rs:298]: Good idea. It can also be used with `--payload-url`

Comment (pothos) [reply src/bin/download_sysext.rs]: Updated it:
```
Options:
  -o, --output-dir  the directory to download the sysext images into
  -n, --target-filename
                    target filename in directory, requires --payload-url or
                    --take-first-match
```

Comment (pothos) [reply src/bin/download_sysext.rs]: I've also removed one option here to keep the number of options low: https://github.com/flatcar/ue-rs/pull/50

**Review by dongsupark — APPROVED** [review]

**Pull request merged**